nixos/modules/services/x11/xserver.nix

```diff
@@ -14,9 +14,6 @@ let
   # Map video driver names to driver packages. FIXME: move into card-specific modules.
   knownVideoDrivers = {
     virtualbox = { modules = [ kernelPackages.virtualboxGuestAdditions ]; driverName = "vboxvideo"; };
-    ati = { modules = with pkgs.xorg; [ xf86videoati ]; };
-    intel = { modules = with pkgs.xorg; [ xf86videointel ]; };
-    modesetting = { modules = []; };
```

Comment:
Now I'm just as confused as you are.  In any case, I'm running the modesetting driver with this change now on my laptop, so it seems to work.